src/main/java/io/quarkus/gizmo/MethodCheckerVisitor.java

```diff
@@ -0,0 +1,223 @@
+package io.quarkus.gizmo;
+
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.Attribute;
+import org.objectweb.asm.Handle;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.TypePath;
+
+public class MethodCheckerVisitor extends MethodVisitor {
```

Comment:
Why is this done as a visitor, rather than just checking that the bytecode is valid when it is generated?

Reply:
imagine you have a code with 
```
public int myfunct(boolean foo, boolean bar) {
 if (foo) return 1
else if (bar) return 0;
else
throw new RuntimeException();
}
```
so you have 2 branchs of return to check indeed.
My first think was to check that last operation was a return and it failed because of that.
You have the exception case too...

Reply:
you can not check just the methodMody, you have to enter in operation to check them.
Or you mean that I check generated code instead of AST ?